axum-macros/src/from_request/attr.rs

```
@@ -0,0 +1,202 @@
+            let via = input.parse::<kw::via>()?;
+            let content;
+            syn::parenthesized!(content in input);
+            content.parse().map(|path| Self::Via { via, path })
+        } else if lh.peek(kw::rejection_derive) {
+            let rejection_derive = input.parse::<kw::rejection_derive>()?;
+            let content;
+            syn::parenthesized!(content in input);
+            content.parse().map(|opt_outs| Self::RejectionDerive {
+                rejection_derive,
+                opt_outs,
+            })
+        } else {
+            Err(lh.error())
+        }
+    }
+}
+
+enum FieldAttr {
+    Via { via: kw::via, path: syn::Path },
+}
+
+impl Parse for FieldAttr {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let lh = input.lookahead1();
+        if lh.peek(kw::via) {
+            let via = input.parse::<kw::via>()?;
+            let content;
+            syn::parenthesized!(content in input);
+            content.parse().map(|path| Self::Via { via, path })
+        } else {
+            Err(lh.error())
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+pub(crate) struct RejectionDeriveOptOuts {
+    debug: Option<kw::Debug>,
+    display: Option<kw::Display>,
+    error: Option<kw::Error>,
+}
+
+impl RejectionDeriveOptOuts {
+    pub(crate) fn derive_debug(&self) -> bool {
+        self.debug.is_none()
+    }
+
+    pub(crate) fn derive_display(&self) -> bool {
+        self.display.is_none()
+    }
+
+    pub(crate) fn derive_error(&self) -> bool {
+        self.error.is_none()
+    }
+}
+
+impl Parse for RejectionDeriveOptOuts {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let mut debug = None;
+        let mut display = None;
+        let mut error = None;
+
+        while !input.is_empty() {
+            input.parse::<Token![!]>()?;
+
+            let lh = input.lookahead1();
+            if lh.peek(kw::Debug) {
+                debug = Some(input.parse()?);
```

Should probably raise an error on multiple assignment here too? Could avoid the extra nesting by moving the assignment or error into the helper function.